tests/cpp-tests/Classes/WindowTest/WindowTest.cpp

```diff
@@ -0,0 +1,62 @@
+#include "WindowTest.h"
+
+USING_NS_CC;
+
+WindowTests::WindowTests() {
+    ADD_TEST_CASE(WindowTestWindowed1);
+    ADD_TEST_CASE(WindowTestWindowed2);
+    ADD_TEST_CASE(WindowTestFullscreen1);
+    ADD_TEST_CASE(WindowTestFullscreen2);
+}
+
+std::string WindowTest::title() const {
+    return "Window Test";
+}
+
+void WindowTestWindowed1::onEnter() {
+    WindowTest::onEnter();
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) || (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX) || (CC_TARGET_PLATFORM == CC_PLATFORM_MAC)
+    GLViewImpl* view = (GLViewImpl*)Director::getInstance()->getOpenGLView();
+    view->setWindowed(480, 320);
+#endif
+}
+
+std::string WindowTestWindowed1::subtitle() const {
+    return "Windowed 480x320";
+}
+
+void WindowTestWindowed2::onEnter() {
+    WindowTest::onEnter();
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) || (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX) || (CC_TARGET_PLATFORM == CC_PLATFORM_MAC)
+    GLViewImpl* view = (GLViewImpl*)Director::getInstance()->getOpenGLView();
+    view->setWindowed(960, 640);
+#endif
+}
+
+std::string WindowTestWindowed2::subtitle() const {
+    return "Windowed 960x640";
+}
+
+void WindowTestFullscreen1::onEnter() {
+    WindowTest::onEnter();
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) || (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX) || (CC_TARGET_PLATFORM == CC_PLATFORM_MAC)
+    GLViewImpl* view = (GLViewImpl*)Director::getInstance()->getOpenGLView();
+    view->setFullscreen();
+#endif
+}
+
+std::string WindowTestFullscreen1::subtitle() const {
+    return "Fullscreen";
+}
+
+void WindowTestFullscreen2::onEnter() {
+    WindowTest::onEnter();
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) || (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX) || (CC_TARGET_PLATFORM == CC_PLATFORM_MAC)
+    GLViewImpl* view = (GLViewImpl*)Director::getInstance()->getOpenGLView();
+    view->setFullscreen(1);
+#endif
+}
+
+std::string WindowTestFullscreen2::subtitle() const {
+    return "Fullscreen Monitor 2 (if present)";
+}
```

Comment:
how about add the macro
```
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) || (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX) || (CC_TARGET_PLATFORM == CC_PLATFORM_MAC)
```
like [this](https://github.com/cocos2d/cocos2d-x/blob/v3/tests/cpp-tests/Classes/controller.cpp#L39). Then the test case only appeared on these platforms, and don't have to add macro in this file.